### PR TITLE
Add mc_weight producer, favor over lhe weight

### PR DESCRIPTION
**Pull request by riga**
This PR adds a new producer `mc_weight` which should be used early on during calibration to create a column with the true MC weight to be used throughout the analysis.

### Background

There are multiple columns that might carry generator weight information:

- `genWeight`
- `Generator.weight`
- `LHEWeight.originalXWGTUP`

According to https://twiki.cern.ch/twiki/bin/view/CMSPublic/WorkBookNanoAOD?rev=99#Weigths the two former columns are identical and this seems true in the 5 different datasets I tested. The next *claim* is that `genWeight` is the product of `LHEWeight.originalXWGTUP` and potential additional weights from the generator chain. However, this is not true for all samples. In some madgraph samples (e.g. `hh_ggf_bbtautau_madgraph`) the lhe weight is some O(1e-2) value, while the `genWeight` is always 1.

To be robust against these bugs in MC, I added a producer that makes a selection between the two options, favoring `genWeight` when it's present and not 1, using the LHE otherwise when existing.

### Changes

- Added the producer
- Adjusted the remaining code to use the `mc_weight` column instead of `LHEWeight.originalXWGTUP`

This should be considered in any downstream analysis.

### CODE REVIEW

File: columnflow/production/mc_weight.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        mc_weight = events.genWeight
+    elif ak.all(events.genWeight == events.LHEWeight.originalXWGTUP):
+        mc_weight = events.genWeight
+    elif ak.all(events.genWeight == 1.0):
+        mc_weight = events.LHEWeight.originalXWGTUP
+    else:
+        raise Exception(
+            f"cannot determine mc_weight from genWeight {events.genWeight} and "
+            f"LHEWeight {events.LHEWeight.originalXWGTUP}",
+        )
```
Comment (mafrahm) [inline columnflow/production/mc_weight.py:48]: the `events.genWeight` and `events.LHEWeight.originalXWGTUP` vector is probably quite large, should we maybe just print a subset, e.g. `events.genWeight[:10]`?

Comment (mafrahm) [reply columnflow/production/mc_weight.py:48]: or even specifically select a sub-set of weights where the conditions are not met

Comment (mafrahm) [reply columnflow/production/mc_weight.py:48]: Something like that maybe?

```
else:
    mask = (events.genWeight != events.LHEWeight.originalXWGTUP) & (events.genWeight != 1)
    raise Exception(
        f"cannot determine mc_weight from genWeight {events.genWeight[mask]} and "
        f"LHEWeight {events.LHEWeight.originalXWGTUP[mask]}",
    )
```

**Review by mafrahm — APPROVED** [review]
Looks good to me, just one small inline comment.

File: columnflow/production/mc_weight.py (hunk 1 above)
Comment (riga) [reply columnflow/production/mc_weight.py:48]: Ah, I was relying on awkward's automatic truncation when printing arrays. Do you think that would be sufficient? :)

Comment (mafrahm) [reply columnflow/production/mc_weight.py:48]: yes, this is probably also sufficient.

**Pull request merged**